crates/core/src/transport/peer_connection.rs

```diff
@@ -359,7 +360,8 @@ mod tests {
             let (tx, rx) = mpsc::channel(1);
             let inbound_msg = tokio::task::spawn(recv_stream(stream_id, message.len() as u64, rx));
             while let Some((_, network_packet)) = receiver.recv().await {
-                let decrypted = PacketData::from(&*network_packet)
+                let network_packet_slice: [u8] = *network_packet;
+                let decrypted = PacketData::new_packet_data(network_packet_slice, network_packet.len()).with_sym_encryption()
```

Comment:
@iduartgomez Having an issue here:

```
Type mismatch [E0308] expected `[u8; <unknown>]`, but found `[u8]` 
```

The problem is that the received network_packet's length isn't known at compile time but I'm not sure how to fix.